### PR TITLE
Add write discover fetch event API 

### PR DESCRIPTION
**Pull request by xiaohansong**
## What
https://github.com/airbytehq/airbyte/issues/21279

To add a new internal API endpoint that workers will call. 

## How
Implement a new Api and hook it up with the existing configRepository logic. Have to implement a converter helper to translate from API model to protocol model.

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
This function will not be used at this moment because workers are still calling DB directly. Thus no user impact.

### CODE REVIEW

**Review by pmossman — COMMENTED** [review]
@xiaohansong I forgot to submit this review yesterday, sorry about that! Left a few comments and suggestions, let me know what you think of them. I can do another pass when needed, just let me know

File: airbyte-api/src/main/openapi/config.yaml [hunk 1]
```diff
@@ -3130,7 +3166,7 @@ components:
     ReleaseStage:
       type: string
       enum:
-        - alpha
+        - alphf
```
[the remark this answers is not in the document]
Comment (xiaohansong) [reply airbyte-api/src/main/openapi/config.yaml:3169]: oops

File: airbyte-api/src/main/openapi/config.yaml [hunk 2]
```diff
@@ -754,6 +754,27 @@ paths:
           $ref: "#/components/responses/NotFoundResponse"
         "422":
           $ref: "#/components/responses/InvalidInputResponse"
+
+  /v1/sources/write_discover_fetch_event:
```
[the remark this answers is not in the document]
Comment (xiaohansong) [reply airbyte-api/src/main/openapi/config.yaml]: done

File: airbyte-server/src/main/java/io/airbyte/server/handlers/helpers/CatalogConverter.java [hunk 3]
```diff
@@ -181,6 +181,30 @@ public static io.airbyte.protocol.models.ConfiguredAirbyteCatalog toProtocolKeep
     return toProtocol(clone);
   }
 
+  public static io.airbyte.protocol.models.AirbyteCatalog toAirbyteCatalogProtocol(
```
[the remark this answers is not in the document]
Comment (xiaohansong) [reply airbyte-server/src/main/java/io/airbyte/server/handlers/helpers/CatalogConverter.java]: yeah agreed. Mike was mentioning we should work with move team to consolidate these two states, but that was a bit out of scope for this PR/tickets.

**Comment by xiaohansong** [comment]
@pmossman Ready for you to take another pass!

**Review by pmossman — APPROVED** [review]
This looks good to me, I think @mfsiega-airbyte should take a quick pass as well to make sure I didn't miss anything as this isn't my most familiar area of code

**Review by mfsiega-airbyte — APPROVED** [review]
LGTM!

**Pull request merged**